### PR TITLE
rtpengine : crash is fixed and codec flags added to doc

### PR DESCRIPTION
**Pull request by ycaner06**
#### Pre-Submission Checklist

- [x] Commit message has the format required by CONTRIBUTING guide
- [x] Commits are split per component (core, individual modules, libs, utils, ...)
- [x] Each component has a single commit (if not, squash them into one commit)
- [x] No commits to README files for modules (changes must be done to docbook files
in `doc/` subfolder, the README file is autogenerated)

#### Type Of Change
- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)

#### Checklist:

- [ ] PR should be backported to stable branches
- [x] Tested changes locally
- [ ] Related to issue #XXXX (replace XXXX with an open issue number)

#### Description

Kamailio crashes when codec-flags value are not setted with codecs. it is fixed.
codec-flags information added to document.
An example is added to document.

### CODE REVIEW

**Comment by miconda** [comment]
@rfuchs - does the patch to the rtpengine module code look ok for you?

File: src/modules/rtpengine/rtpengine.c [hunk 1]
```diff
@@ -1992,6 +1992,10 @@ static int parse_flags(struct ng_flags_parse *ng_flags, struct sip_msg *msg, enu
 		if (str_key_val_prefix(&key, "transcode", &val, &s)
 				|| str_key_val_prefix(&key, "codec-transcode", &val, &s))
 		{
+			if(s.len<=0 && s.s==NULL){
```
Comment (rfuchs) [inline src/modules/rtpengine/rtpengine.c]: There's actually a much better solution to this, which is to change `str_key_val_prefix()`, which is where the bug actually is.

The function should read:

```
static inline int str_key_val_prefix(const str *p, const char *q, const str *v, str *out) {
	if (str_eq(p, q)) {
		if (!v->s || !v->len)
			return 0;
		*out = *v;
		return 1;
	}
...
`

Then all the explicit null checks here and below can be removed.

Comment (ycaner06) [reply src/modules/rtpengine/rtpengine.c]: I avoid changing function because i afraid that breaking other parts. At monday , i can change it as you wish.

**Comment by rfuchs** [comment]
Looks good to me now. @miconda do you prefer to have these commits squashed or merge as they are now?

**Comment by miconda** [comment]
@rfuchs - up to you, do you it as you prefer -- usually we squash if a commit is changing or is an improvement of the previous commit.

**Pull request merged**